### PR TITLE
Add EIP-5659: Social Media URI Propagation Event

### PR DESCRIPTION
**Pull request by TimDaub**
Heavily inspired by EIP-3722 of @auryn-macmillan

### CODE REVIEW

File: EIPS/eip-unassigned.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+---
+eip: unassigned
```
Comment (xinbenlv) [inline EIPS/eip-unassigned.md:2]:
```suggestion
eip: 5659
```

Comment (xinbenlv) [reply EIPS/eip-unassigned.md:2]: Also don't forget to update the file name.

File: EIPS/eip-5659.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+created: 2022-09-15
+---
+
+## Abstract
+
+We introduce a minimal on-chain primitive to emit URIs using Ethereum's event log infrastructure.
+
+## Motivation
+
+The modern social network's base layer is distributing content behind Universal Resource Identifiers as defined by RFC 3986. Any web resource can be identified by an URI or a URL and so upgrading Ethereum with a standard way of distributing content is useful.
```
Comment (SamWilsn) [inline EIPS/eip-5659.md:19]: You haven't convinced me that this EIP needs to exist. I think I'd like to see some potential use cases, both for publishers and for event consumers.

File: EIPS/eip-5659.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+discussions-to: https://ethereum-magicians.org/t/social-media-uri-propagation-event/10893
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-09-15
+---
+
+## Abstract
+
+We introduce a minimal on-chain primitive to emit URIs using Ethereum's event log infrastructure.
```
Comment (SamWilsn) [inline EIPS/eip-5659.md:15]: We like terse abstracts, but this is perhaps a touch too short. Perhaps include a brief summary of what kinds of URIs you expect people to publish?

File: EIPS/eip-5659.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+The modern social network's base layer is distributing content behind Universal Resource Identifiers as defined by RFC 3986. Any web resource can be identified by an URI or a URL and so upgrading Ethereum with a standard way of distributing content is useful.
+
+## Specification
+
+```solidity
+contract Propagator {
+  event NewPost(address indexed author, string indexed uri);
+  function submit(string calldata uri) external {
+    emit NewPost(msg.sender, uri);
+  }
```
Comment (SamWilsn) [inline EIPS/eip-5659.md:28]: Is this implementation the only valid implementation? If not, you should probably switch this to an interface and let implementers decide how to implement it.

File: EIPS/eip-5659.md (hunk 2 above)
Comment (TimDaub) [reply EIPS/eip-5659.md:19]: ok, how would you frame this in the context of an EIP though? E.g. I feel like pointing out the countless problems of web2 content distribution could read too political

Comment (SamWilsn) [reply EIPS/eip-5659.md:19]: I don't think you need to bash web2 to make a case for this EIP. What URIs do you expect to publish? Who do you expect to consume them? What applications are difficult to implement without this standard? These questions, answered in your Motivation section, would make for a more compelling proposal.

**Pull request closed without merging**